build: add karma tests for render3

mhevery opened the pull request.

jasonaden committed:
fix(router): fix wildcard route with lazy loaded module (again) (#18139)

Closes #13848

Description:
We doesn't handle children of wildcard route properly link. It's always an empty array.

Created from #13851

PR Close #18139

ocombe committed:
fix(common): handle JS floating point errors in percent pipe (#20329)

Fixes #20136
PR Close #20329

mhevery committed:
feat(core): Moving Renderer3 into @angular/core (#20855)

PR Close #20855

mhevery committed:
refactor(core): store locals in main array in rederer3 (#20855)

PR Close #20855

marclaval committed:
feat(core): add renderer factory in render3 (#20855)

PR Close #20855

marclaval committed:
perf: add large table and deep tree benchmarks for render3 (#20855)

PR Close #20855

kara committed:
refactor(core): store directive defs in static data (#20855)

PR Close #20855

pkozlowski-opensource committed:
feat: support queries for elements with local names (#20855)

PR Close #20855

marclaval committed:
perf: add benchmarks for IV and render3 with functions (tree) (#20855)

PR Close #20855

marclaval committed:
perf: use style.setProperty instead of setAttribute in render3 and iv benchmarks (#20855)

PR Close #20855

kara committed:
docs(core): add comments to assert functions (#20855)

PR Close #20855

kara committed:
docs(core): add comments to node_manipulation functions (#20855)

PR Close #20855

kara committed:
docs(core): document di and minor renames (#20855)

PR Close #20855

kara committed:
docs(core): add more comments to di and fix formatting (#20855)

PR Close #20855

kara committed:
refactor(core): renamed and split out interfaces (#20855)

PR Close #20855

Teamop committed:
docs(changelog): fix typo in 5.1.1 (#21007)

PR Close #21007

mhevery committed:
style: fix formatting errors (#20855)

PR Close #20855

mhevery committed:
style: fix formatting errors (#20855)
PR Close #20855

kara committed:
docs(core): add docs to instructions, minor renames (#20855)

PR Close #20855

kara committed:
refactor(core): rename instructions for consistency (#20855)

PR Close #20855

kara committed:
docs(core): add missing docs to component and fix formatting (#20855)

PR Close #20855

pkozlowski-opensource committed:
refactor: remove import circular dependencies (#20855)

This PR fixes a circular dependency among those files in Renderer3:
`query` -> `di` -> `instructions` -> `query` -> ...

Looking at the above dependencies the `di` -> `instructions` import is
a problematic one. Previously `di` had an import from `instructions`
since we can known about "current node" only in `instructions`
(and we need "current node" to create node injector instances).

This commit refactors the code in the way that functions in the
`di` file don't depend on any info stored module-global variables
in `instructions`.

PR Close #20855

marclaval committed:
test(core): animation renderer factory in render3 (#20855)

PR Close #20855

marclaval commented on packages/core/test/render3/BUILD.bazel:
```diff
@@ -13,21 +13,30 @@ ts_library(
 
 jasmine_node_test(
     name = "render3",
+    srcs = ["load_domino.js"],
```
Isn't `load_domino.ts` removed in this PR?

mhevery replied:
that was an accidental deletion

chuckjaz committed:
fix(compiler): report an error for recursive module references

Modules that directly or indirectly export or imported themselves
reports an error instead of generating a stack fault.

Fixes: #19979

mhevery committed:
build(core): remove `main()` from specs (#21053)

`main()` function used to be needed to support dart, since dart
Does not allow top level statements. Since we no longer use dart
The need for `main()` has been removed.

In preparation for `Basel` and standardized way of running tests
we are removing `main()`

PR Close #21053

mhevery committed:
build(core): add bazel test targets for core (#21053)

- Add tests target for `test`, `test_node_only` and `test_web` in `core` package.
- Created a `_testing_init` pseudo package where bootstrap code for tests is kept.
- Moved `source_map_util` from `test` to `testing` so to prevent circular dependency.
- Removed `visibility:public` for testing `BUILD` packages.

PR Close #21053

mhevery committed:
build: remove `main()` from specs (#21053)

PR Close #21053

mhevery committed:
build: add bazel test rules for more packages (#21053)

PR Close #21053

mhevery committed:
build: add bazel test rules for remainder of packages (#21053)

PR Close #21053

mhevery committed:
build(router): enable bazel tests for router (#21053)

Bazel runs on newer version of RxJs than is installed in Yarn. The never version subclasses `EmptyError` in a different way which fails the `instanceof` check. This change makes the `instanceof` check more robust with respect to `EmptyError`.

PR Close #21053

mhevery committed:
build(upgrade): enable bazel tests for upgrade (#21053)

Added missing angular-mocks dependency.

PR Close #21053

mhevery committed:
build(service-worker): enable karma bazel test for service-worker (#21053)

Corrected the environment detection code which was incorretly throwing exception in the browser if `require` function was found.

PR Close #21053

mhevery committed:
build(compiler-cli): exclude command line scripts from compilation (#21053)

PR Close #21053

mhevery committed:
build(platform-browser): exclude node incompatible tests from :test target. (#21053)
PR Close #21053

pkozlowski-opensource committed:
feat(core): support 'read' option for ngIvy queries (#20855)

PR Close #20855

mhevery committed:
build: move _testing_init into tools; limit web_test concurrency (#21053)

PR Close #21053

mhevery committed:
build: add istruction for running and debugging tests to `BAZEL.md` (#21053)

PR Close #21053

mhevery committed:
build: upgrade circle/build into large class (more ram) (#21053)

PR Close #21053

mhevery committed:
build: yarn buildifier (#20855)

PR Close #20855

mhevery committed:
build: fix benchmarks for render3 (#20855)

PR Close #20855

kapunahelewong committed:
docs(forms): add text about min() and max() as functions (#21110)

PR Close #21110

mhevery committed:
build: fix circular dep between interface and l_node by merging (#20855)

PR Close #20855

bitcldr committed:
docs(service-worker): fix word wrap (#21114)

The fix removes space between 'c' and 'aches' in docs

PR Close #21114

alxhub committed:
ci(router): update the public API guard for the router

This fixes a badly applied revert earlier.

alxhub committed:
docs: add changelog for 5.1.2

alxhub committed:
release: cut the 5.1.0-beta.1 release

alxhub committed:
docs: add changelog for 5.2.0-beta.1

alexeagle committed:
build: update pullapprove: (#21117)

- remove ex-team members
- allow author to approve their own change
- move more bazel files under the bazel group

PR Close #21117

matsko committed:
fix(animations): avoid infinite loop with multiple blocked sub triggers (#21119)

This patch fixes animations so that if multiple sub @triggers are used
and are blocked by a parent animation then the engine will not lead
itself into an infinite loop.

PR Close #21119

mprobst committed:
fix(animations): renaming issue with DOMAnimation. (#21125)

Closure Compiler renames all properties that are "internal" to the
program. `DOMAnimation` however is external, it is a browser API, so its
fields must not be renamed.

This change marks `DOMAnimation` as external using `declare interface`,
which will cause Closure Compiler to back off and prevent renaming of
any of its fields.

PR Close #21125

ocombe committed:
build(common): generate ts declarations for i18n locale files (#21127)

Fixes #21120
PR Close #21127

alexeagle committed:
build: fix pullapprove (#21140)

Currently it gives a green status if I edit a file I'm an owner of,
even without anyone else's approval.

PR Close #21140

IgorMinar committed:
build: make umd.min.js source map paths relative (#21147)

I'm not quite sure how to test this since we don't have any infrastructure for these kinds of tests.
I did verify the fix manually though.

Fixes #15740

PR Close #21147

chuckjaz committed:
build: upgrade to TypeScript 2.6 (#21144)

Fixes #20653

PR Close #21144

mhevery committed:
build: add karma tests for render3
